mne/io/persyst/persyst.py

```diff
@@ -0,0 +1,443 @@
+# Authors: Adam Li <adam2392@gmail.com>
+#
+# License: BSD (3-clause)
+import os
+import os.path as op
+import time
+from datetime import datetime, timezone
+
+import numpy as np
+
+from ..base import BaseRaw
+from ..constants import FIFF
+from ..meas_info import create_info
+from ...annotations import Annotations
+from ...utils import logger, verbose, fill_doc, warn
+
+# number of bytes for 16 and 32 bit binary .dat file
+_fmt_byte_dict = dict(short=2, long=4)
+
+
+@fill_doc
+def read_raw_persyst(fname, preload=False, verbose=None):
+    """Reader for a Persyst (.lay/.dat) recording.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the Persyst header (.lay) file.
+    %(preload)s
+    %(verbose)s
+
+    Returns
+    -------
+    raw : instance of RawPersyst
+        A Raw object containing Persyst data.
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    return RawPersyst(fname, preload, verbose)
+
+
+def _open(fname):
+    return open(fname, 'r', encoding='latin-1')
+
+
+@fill_doc
+class RawPersyst(BaseRaw):
+    """Raw object from a Persyst file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the Persyst header (.lay) file.
+    %(preload)s
+    %(verbose)s
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+
+    @verbose
+    def __init__(self, fname, preload=False, verbose=None):
+        logger.info('Loading %s' % fname)
+
+        if not fname.endswith('.lay'):
+            fname = fname + '.lay'
+        curr_path, lay_fname = op.dirname(fname), op.basename(fname)
+        if not op.exists(fname):
+            raise FileNotFoundError(f'The path you specified, '
+                                    f'"{lay_fname}",does not exist.')
+
+        # takes ~8 min for a 1.5GB file
+        start_time = time.time()
+
+        # sections and subsections currently unused
+        keys, data, sections, subsections = _read_lay_contents(fname)
+
+        # these are the section headers in the Persyst file layout
+        fileinfo_dict = dict()
+        channelmap_dict = dict()
+        sampletimes_dict = dict()
```

Comment:
Yeah sampletimes seems redundant, and I included it in here because it was a field, but you can obtain the same info via the `testtime` and `testdate`.